test/http_mock.dart

```diff
@@ -39,8 +40,14 @@ class MockHttpHeaders implements HttpHeaders {
   ContentType contentType;
 
   @override
-  void set(String name, Object value) {
-    name = name.toLowerCase();
+  void set(String name, Object value, {bool preserveHeaderCase = false}) {
+    // The casing of Header name is determined by the latest `set` call.
+    if (preserveHeaderCase) {
+      _originalHeaderName[name.toLowerCase()] = name;
+    } else {
+      name = name.toLowerCase();
+      _originalHeaderName[name] = name;
+    }
     _headers.remove(name);
```

Comment:
We'd want the lowercase name here even if `preserveHeaderCase` is true.

Reply:
Actually I'm not confident with this behavior change.
Is this the behavior this package wants? If not, I just need to update the signature. Or I can do that whenever there is a request.
What do you think?